src/main/java/com/ibm/disni/verbs/RdmaConnParam.java

```diff
@@ -221,4 +225,23 @@ public int getQp_num() {
 	public void setQp_num(int qp_num) throws IOException {
 		throw new IOException("Operation currently not supported");
 	}
+
+
+	public void writeBack(ByteBuffer buffer) {
+		buffer.order(ByteOrder.LITTLE_ENDIAN);
+		buffer.putLong(private_data_addr);
+		buffer.order(ByteOrder.nativeOrder());
+		buffer.put(private_data_len);
+		buffer.put(responder_resources);
+		buffer.put(initiator_depth);
+		buffer.put(flow_control);
+		buffer.put(retry_count);
+		buffer.put(rnr_retry_count);
+		buffer.put(srq);
+		buffer.put((byte)0);
```

Comment:
cq_num is u32 not u8, so this should be putInt(0)

Reply:
here I do not agree. Actually, buffer.put((byte)0); is a padding. cq_num field is 8 byte aligned.  To add cq_num the code should be: buffer.put(srq); buffer.put((byte)0); buffer.putInt(cq_num);
I did not add cq_num as it is ignored anyway and does not cause any problems.

Reply:
Makes sense.